streams/src/test/java/org/apache/kafka/streams/integration/KStreamCogroupIntegrationTest.java

```diff
@@ -0,0 +1,489 @@
+    private static final List<Input<Long, String>> INPUTS = Arrays.asList(
+            new Input<>(INPUT_TOPIC_1, 10L, new KeyValue<>(1L, "a")),
+            new Input<>(INPUT_TOPIC_2, 10L, new KeyValue<>(2L, "a")),
+            new Input<>(INPUT_TOPIC_3, 11L, new KeyValue<>(1L, "a")),
+            new Input<>(INPUT_TOPIC_1, 11L, new KeyValue<>(1L, "b")),
+            new Input<>(INPUT_TOPIC_2, 12L, new KeyValue<>(2L, "b")),
+            new Input<>(INPUT_TOPIC_3, 12L, new KeyValue<>(1L, "b")),
+            new Input<>(INPUT_TOPIC_1, 20L, new KeyValue<>(2L, "c")),
+            new Input<>(INPUT_TOPIC_2, 20L, new KeyValue<>(1L, "c")),
+            new Input<>(INPUT_TOPIC_3, 21L, new KeyValue<>(2L, "c")),
+            new Input<>(INPUT_TOPIC_1, 21L, new KeyValue<>(2L, "a")),
+            new Input<>(INPUT_TOPIC_2, 22L, new KeyValue<>(1L, "a")),
+            new Input<>(INPUT_TOPIC_3, 22L, new KeyValue<>(2L, "a")),
+            new Input<>(INPUT_TOPIC_1, 16L, new KeyValue<>(2L, "b")),
+            new Input<>(INPUT_TOPIC_2, 16L, new KeyValue<>(1L, "b")),
+            new Input<>(INPUT_TOPIC_3, 17L, new KeyValue<>(2L, "b")),
+            new Input<>(INPUT_TOPIC_1, 17L, new KeyValue<>(1L, "c")),
+            new Input<>(INPUT_TOPIC_2, 18L, new KeyValue<>(2L, "c")),
+            new Input<>(INPUT_TOPIC_3, 18L, new KeyValue<>(1L, "c"))
+        );
+
+    @BeforeClass
+    public static void setupConfigs() throws Exception {
+        PRODUCER_CONFIG.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+        PRODUCER_CONFIG.put(ProducerConfig.ACKS_CONFIG, "all");
+        PRODUCER_CONFIG.put(ProducerConfig.RETRIES_CONFIG, 0);
+        PRODUCER_CONFIG.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, LongSerializer.class);
+        PRODUCER_CONFIG.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+
+        CONSUMER_CONFIG.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+        CONSUMER_CONFIG.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        CONSUMER_CONFIG.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, LongDeserializer.class);
+        CONSUMER_CONFIG.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
+
+        STREAMS_CONFIG.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+        STREAMS_CONFIG.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        STREAMS_CONFIG.put(StreamsConfig.STATE_DIR_CONFIG, TestUtils.tempDirectory().getPath());
+        STREAMS_CONFIG.put(StreamsConfig.KEY_SERDE_CLASS_CONFIG, Serdes.Long().getClass());
+        STREAMS_CONFIG.put(StreamsConfig.VALUE_SERDE_CLASS_CONFIG, Serdes.String().getClass());
+        STREAMS_CONFIG.put(StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, 0);
+    }
+
+    @Test
+    public void testCogroup() throws InterruptedException, ExecutionException {
+        final int testNumber = TEST_NUMBER.getAndIncrement();
```

Comment:
Looks like you can do these 7 lines in a `@Before` method and not repeat it in all of your tests